docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+"number", "string" and "object" are used to denote the types as
+defined in [RFC8259]_. The term "array" is also used as defined in
+[RFC8259]_, except where qualified as "Zarr array". Following
+[RFC8259]_, this section also describes an object as a set of
+name/value pairs. This section also defines how metadata documents are
+encoded for storage.
+
+
+Only the top level metadata document ``zarr.json`` is guaranteed to be of JSON
+type, and can be used to define other formats for array-level and group-level
+metadata documents. In the case where non-JSON metadata documents are used in a
+Zarr hierarchy, the following sections on group and array level metadata are
+non-normative, but other metadata formats are expected to define some
+equivalence relations with the JSON documents.
+
+
+Entry point metadata
+--------------------
+
+Each Zarr hierarchy must have an entry point metadata document, which
+provides essential information regarding the format version being
+used, the encoding being used for group and array metadata, and any
+extensions that affect the layout or interpretation of data
+in the store.
+
+The entry point metadata document must contain a single object
+containing the following names:
+
+``zarr_format``
+^^^^^^^^^^^^^^^
+
+    A string containing the URI of the Zarr core
+    specification that defines the metadata format. For Zarr
+    hierarchies conforming to this specification, the value must be
+    the string "https://purl.org/zarr/spec/core/3.0".
+
+    Implementations of this specification may assume that the final path
+    segment of this URI ("3.0") represents the core specification version
+    number, where "3" is the major version number and "0" is the minor
+    version number. Implementations of this specification may also assume
+    that future versions of this specification that retain the same major
+    versioning number ("3") will be backwards-compatible, in the sense
+    that any new features added to the specification can be safely
+    ignored. In other words, if the major version number is "3",
+    implementations of this specification may read and interpret metadata
+    as defined in this specification, ignoring any name/value pairs
+    where the name is not defined here. See also the `stability policy`_.
+
+    Note that this value is given as a URI rather than as a simple
+    version number string to help with discovery of this
+    specification.
+
+``metadata_encoding``
+^^^^^^^^^^^^^^^^^^^^^
+
+    A string containing the URI pointing to a document describing the method
+    used for encoding group and array metadata documents.
+
+    For document using the default JSON encoding and format describe in this document
+    then the value must be ``"https://purl.org/zarr/spec/core/3.0``.
+
+``metadata_key_suffix``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+    A string containing a suffix to add to the metadata keys when saving into
+    the store. By default ``".json"``.
+
+    .. note::
```

Comment:
I am not quite sure what's the point of this note. How does the suffix defined in the metadata help non-zarr-aware tools?

Reply:
I'm also not sure this extra parameter is really worth its added complexity.  In practice if almost everyone uses ".json" then some tools may come to rely on that and break if a different suffix is used.

Reply:
The note is probably about the motivation for the change between v2 and v3: When having a `.zattrs` file on aws with v2, it had the octet-stream mimetype and was downloaded by browsers by default.

Instead of dropping the suffix field, I propose to drop `metadata_encoding` instead, and use the `metadata_key_suffix` to define an (associated) encoding instead. This would just be used by extensions, but would make it easier to actually change the suffix and not have `.json` files with cbor encoding. See 75c44ce in PR #163. I'm resolving this here for now, but please feel free to comment on the PR directly or raise a separate issue about it if you feel that this does not resolve it properly.

Reply:
Correction: I combined the `metadata_key_suffix` into the `metadata_encoding` field and specified it as an extension in #171, but there's also the proposal to drop them completely (#174).